# Coursera "Applied Data Science Capstone" Project -- Week 3 -- Part 1

## Segmenting and Clustering Neighborhoods in Toronto

#### Firstly, installing required libraries
(For Part 1)

In [3]:
from bs4 import BeautifulSoup
import requests
import pandas as pd

#### Secondly, scrape data set from Wikipedia website

!! We see that, in wikipedia the needs of our data exist as a "table", it is easy to find the table to develop the dataframe with this table data 

In [12]:
# read the webpage from the wiki
url='https://en.wikipedia.org/wiki/List_of_postal_codes_of_Canada:_M'
response = requests.get(url)
#Process and convert html data
data = response.text
soup = BeautifulSoup(data,'html.parser')
wiki_table=soup.find('table')
#develop dataframe
df = pd.read_html(str(wiki_table))[0]
df.head()

0             1                 2
0  Postcode       Borough     Neighbourhood
1       M1A  Not assigned      Not assigned
2       M2A  Not assigned      Not assigned
3       M3A    North York         Parkwoods
4       M4A    North York  Victoria Village

#### Thirdly, prepare the dataframe in pursuant of instructions

We need to drop first column and arrange the columns names

In [5]:
# Drop the first column
df.drop(0,inplace=True)
#Rename the columns names
df.columns = ['PostalCode','Borough','Neighborhood']
df.head()

PostalCode           Borough      Neighborhood
1        M1A      Not assigned      Not assigned
2        M2A      Not assigned      Not assigned
3        M3A        North York         Parkwoods
4        M4A        North York  Victoria Village
5        M5A  Downtown Toronto      Harbourfront

Cleaning data as ignoring the rows which the "Borough" contains 'Not assigned'

In [8]:
# Remove "Borough" with 'Not assigned' values
df2=df[df['Borough'].str.contains("Not assigned") == False].reset_index()
df2.head()

index PostalCode           Borough      Neighborhood
0      3        M3A        North York         Parkwoods
1      4        M4A        North York  Victoria Village
2      5        M5A  Downtown Toronto      Harbourfront
3      6        M5A  Downtown Toronto       Regent Park
4      7        M6A        North York  Lawrence Heights

Combine the rows into one row (and seperate with comma) which PostalCode and Borough as same value

In [9]:
df3= df2.groupby(['PostalCode', 'Borough'])['Neighborhood'].apply(', '.join).reset_index()
df3.head()

PostalCode      Borough                            Neighborhood
0        M1B  Scarborough                          Rouge, Malvern
1        M1C  Scarborough  Highland Creek, Rouge Hill, Port Union
2        M1E  Scarborough       Guildwood, Morningside, West Hill
3        M1G  Scarborough                                  Woburn
4        M1H  Scarborough                               Cedarbrae

Get the number of rows of dataframe

In [11]:
df3.shape

(103, 3)

we get 103 unique data

Save the file as csv file for using in part 2

In [16]:
#Save data as'Capstone_part1.csv'
df3.to_csv('Capstone_part1.csv')
print('Successfully Saved!')

Successfully Saved!
